src/sql/workbench/api/common/extHostErrorDiagnostics.ts

```diff
@@ -0,0 +1,86 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the Source EULA. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import * as azdata from 'azdata';
+import { IMainContext } from 'vs/workbench/api/common/extHost.protocol';
+import { Disposable } from 'vs/workbench/api/common/extHostTypes';
+import {
+	ExtHostErrorDiagnosticsShape,
+	MainThreadErrorDiagnosticsShape,
+} from 'sql/workbench/api/common/sqlExtHost.protocol';
+import { values } from 'vs/base/common/collections';
+import { SqlMainContext } from 'vs/workbench/api/common/extHost.protocol';
+
+export class ExtHostErrorDiagnostics extends ExtHostErrorDiagnosticsShape {
+	private _handlePool: number = 0;
+	private _proxy: MainThreadErrorDiagnosticsShape;
+	private _providers: { [handle: number]: DiagnosticsWithMetadata } = {};
+
+	constructor(mainContext: IMainContext) {
+		super();
+		this._proxy = mainContext.getProxy(SqlMainContext.MainThreadErrorDiagnostics);
+	}
+
+	// PUBLIC METHODS //////////////////////////////////////////////////////
+	// - MAIN THREAD AVAILABLE METHODS /////////////////////////////////////
+	public override $handleError(handle: number, errorCode: number, errorMessage: string, connection: azdata.connection.ConnectionProfile): Thenable<boolean> {
+		return this._withProvider(handle, (provider: azdata.diagnostics.ErrorDiagnostics) => provider.handleError(errorCode, errorMessage, connection));
```

Comment:
if no provider is found, we should return the default value 'false' here, right?

Reply:
I'm following the code format established by extHostAccountManagement and extHostResourceProvider which does not do this.